engine/class_modules/sc_mage.cpp

```diff
@@ -3498,10 +3493,13 @@ struct arcane_barrage_t final : public dematerialize_spell_t
     {
       p()->trigger_clearcasting( 1.0, 0_ms );
       p()->trigger_arcane_charge( arcane_soul_charges );
+      p()->buffs.arcane_soul_counter->trigger( p()->buffs.arcane_soul->remains() );
```

Comment:
The counter doesn't last that long, it can actually expire before Arcane Soul does.